sentry/src/db/analytics.rs

```diff
@@ -0,0 +1,95 @@
+use crate::db::DbPool;
+use crate::Session;
+use bb8::RunError;
+use chrono::Utc;
+use primitives::analytics::{AnalyticsQuery, AnalyticsResponse};
+
+pub async fn get_analytics(
+    query: AnalyticsQuery,
+    channels: Option<String>,
+    sess: Option<&Session>,
+    pool: &DbPool,
+    is_advertiser: bool,
+    skip_publisher_filter: bool,
```

Comment:
Better to fix those 2 parameters as they, for what I can see, are bounded to the `Session` and we do an `unwrap` on the session, which might be wrongly passed. I feel that this approach resembles more JS style than what we can do with Rust.

We can use a Struct or Enum to represent those filters and include the Session and handle the cases inside.

Also `skip_publisher_filter`, as general rule of thumb for me, should be renamed to something more clean for the API like `filter_publisher`.

Reply:
`Better to fix those 2 parameters as they` ... I don't get this

Reply:
in a nutshell, if I pass `None` & `is_advertiser: false`, this will blow up. We need a better API for this function.